src/routes/swish.ts

```diff
@@ -0,0 +1,94 @@
+import bodyParser from "body-parser";
+import express, { RequestHandler } from "express";
+import ipRangeCheck from "ip-range-check";
+import config from "../config";
+import * as swish from "../custom_modules/swish";
+
+const router = express.Router();
+const jsonBody = bodyParser.json();
+
+const swishWhitelistMiddleware: RequestHandler = (req, res, next) => {
+  if (!ipRangeCheck(req.ip, config.swish_whitelist)) {
+    console.warn(`Swish request from non-whitelisted IP: ${req.ip}`);
+    res.sendStatus(403);
+    return;
+  }
+  next();
+};
+
+/**
+ * @openapi
+ * /swish/orders/{id}:
+ *    get:
+ *      tags: [Swish]
+ *      description: Fetches a Swish order by id
+ *    parameters:
+ *      - in: path
+ *        name: id
+ *        required: true
+ *        description: Numeric ID of the order to fetch.
+ *        schema:
+ *          type: integer
+ *    responses:
+ *      200:
+ *        description: Swish order
+ *        content:
+ *          application/json:
+ *            schema:
+ *              - type: object
+ *                properties:
+ *                  ID:
+ *                    type: integer
+ *                  status:
+ *                    type: string
+ */
+router.get("/orders/:id", async (req, res, next) => {
```

Comment:
Should be protected by isAdmin check?

Reply:
Good question! The purpose of this endpoint is for the client to be able to poll the status of the order. So the flow for making a swish payment would be:

1. POST /donations/register with swish payment method. An id is returned.
2. Poll to GET /swish/order/:id until order status is paid

Although I realise now, the donation registration route does not return the swish order id. Aside that, I'm not sure if the route needs any authentication. Perhaps the donation route could return a JWT token that the client would need to use for authentication, but that would require some work.

I'll have a look on either returning the swish order id, or changing the route to accept a KID parameter instead.

Reply:
Ah, I see!

Well, as long as the endpoint doesn't return anything sensitive, I guess this is fine?

Reply:
Yes, I think so too!